docs/index.md

```diff
@@ -0,0 +1,303 @@
+
+- require an `id` &mdash; this should be unique to your application
+- optionally support a `schema` &mdash; if provided, this should be a [Standard Schema](https://standardschema.dev) compatible schema instance, such as a [Zod](https://zod.dev) or [Effect](https://effect.website/docs/schema/introduction/) schema
+
+The Electric collection then also requires two additional options:
+
+- `streamOptions` &mdash; the Electric [ShapeStreamOptions](https://electric-sql.com/docs/api/clients/typescript#options) that define the [Shape](https://electric-sql.com/docs/guides/shapes) to sync into the collection; this includes the
+  - `url` to your sync engine; and
+  - `params` to specify the `table` to sync and any optional `where` clauses, etc.
+- `primaryKey` &mdash; identifies the primary key for the rows being synced into the collection
+
+When you create the collection, sync starts automatically.
+
+Electric shapes allow you to filter data using where clauses:
+
+```ts
+export const myPendingTodos = createElectricCollection<Todo>({
+  id: 'todos',
+  schema: todoSchema,
+  streamOptions: {
+    url: 'https://example.com/v1/shape',
+    params: {
+      table: 'todos',
+      where: `
+        status = 'pending'
+        AND
+        user_id = '${user.id}'
+      `
+    }
+  },
+  primaryKey: ['id']
+})
+```
+
+> [!Tip] Shapes vs Queries
+> TanStack DB de-couples the data you sync into a collection from the data you bind to a component. Shape where clauses are used to filter the data you sync into collections. [Live queries](#live-queries) are used to bind data to components.
+>
+> Live queries are much more expressive than shapes, allowing you to query across collections, join, aggregate, etc. This allows you to de-normalise the data you bind to a component.
+>
+> Shapes are normalised: they just contain filtered database tables.
+
+If you need more control over what data syncs into the collection, Electric allows you to [use your API](https://electric-sql.com/blog/2024/11/21/local-first-with-your-existing-api#filtering) as a proxy to both authorise and filter data.
+
+See the [Electric docs](https://electric-sql.com/docs/intro) for more information.
+
+### QueryCollection
+
+[TanStack Query](https://tanstack.com/query) fetches data using managed queries. Use `createQueryCollection` to fetch data into a collection using TanStack Query:
+
+```ts
+const todoCollection = createQueryCollection({
+  queryKey: ['todoItems'],
+  queryFn: async () => fetch('/api/todos'),
+  getPrimaryKey: (item) => item.id
+})
+```
+
+The collection will be populated with the query results.
+
+### base Collection
+
+There is a base `Collection` class in [`../packages/optimistic/src/collection.ts`](../packages/optimistic/src/collection.ts). You can use this directly or as a base class for implementing your own collection types.
+
+See the existing implementations in [`../packages/db-collections`](../packages/db-collections) for reference.
+
+
+## Live Queries
```

Comment:
@samwillis perhaps there's more you'd like to add here?

For example, the non-hook query interface? Also I think the "it's collections all the way down" concept is powerful, i.e.: that queries can be composed. Perhaps we can surface that in the usage docs here?

Reply:
Yeah, it's very powerful and unique. 

Two ideas I've liked of the creating materialized views for easier querying in the app and cloning a collection to scope optimistic updates to part of the app.